cartoframes/data/dataset.py

```diff
@@ -181,7 +183,7 @@ def dataset_info(self):
 
         return self._strategy.dataset_info
 
-    def update_dataset_info(self, privacy=None, name=None):
+    def update_dataset_info(self, privacy=None, name=None, table_name=None):
```

Comment:
🤔 are not `name` and `table_name` the same thing?

Reply:
If I am not wrong:
- table_name: table name in postgres
- name: dataset name in CARTO

Reply:
yes, but I guess both are related.

Reply:
Yes, if you change the dataset name, the table changes too

Reply:
We also need to change the table name of the Dataset instance

Reply:
I realized they have 2 different meanings here and we need both of them:
- name: is the name of the dataset in CARTO (also the table as you said before)
- table_name: when the query is using several tables, this prop is used to select what table do we want to update

Example: we have a query using 2 tables (table1 and table2) and we want to rename the first one to table3:
```
ds = Dataset('SELECT * FROM table1 LEFT JOIN table2 ON ...')
ds.update_dataset_info(table_name='table1' name='table3')
```
Why!! I am so bad for naming :cry: 

Ok, my proposal:
- name ---> `remote_table_name` or `metadata_name` (should we change `privacy` to `remote_privacy` or `metadata_privacy` ?)
- table_name ---> `table_name`

Reply:
I wonder, should we even offer the ability to update tables involved in a query. I like the flow in `Dataset(query).delete()` where we just say 'sorry can't do that'.

Another question: Does the `CDB_QueryTablesText` always return the table names in the same order as they appear in the query?

Reply:
> I wonder, should we even offer the ability to update tables involved in a query. I like the flow in Dataset(query).delete() where we just say 'sorry can't do that'.

:thinking: Thank you for opening my mind. I think you are right because in any case, it will be a very rare use case. And in the end, as we said about the delete method if you want to change something from a table, use a table, not a query.

> Does the CDB_QueryTablesText always return the table names in the same order as they appear in the query?

I don't think so, but I can not ensure it showing the code https://github.com/CartoDB/cartodb-postgresql/blob/362af5e6a0792ce65e8a842cdc1c0dd36d6da6ad/scripts-available/CDB_QueryTables.sql

Reply:
I guess it doesn't matter about the return order because the user specifies the table name. For some reason I thought they would use the index/order, but that's wrong.

Reply:
Done, In the same way as delete, dataset_info method still returns an error